src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala

```diff
@@ -103,5 +107,37 @@ trait LdapSupport extends DirectorySubjectNameSupport {
   def evictIsMemberOfCache(subject: WorkbenchSubject): IO[Unit] =
     IO.pure(memberOfCache.remove(subject))
 
+  def ldapLoadGroup(groupName: WorkbenchGroupName): IO[Option[BasicWorkbenchGroup]] = {
+    val res = for {
+      entry <- OptionT(executeLdap(IO(ldapConnectionPool.getEntry(groupDn(groupName)))).map(Option.apply))
+      r <- OptionT.liftF(IO(unmarshalGroupThrow(entry)))
+    } yield r
+
+    res.value
+  }
+
+  def ldapLoadGroups(groupNames: Set[WorkbenchGroupName]): IO[Stream[BasicWorkbenchGroup]] = {
+    val filters = groupNames.grouped(batchSize).map(batch => Filter.createORFilter(batch.map(g => Filter.createEqualityFilter(Attr.cn, g.value)).asJava)).toSeq
+
+    executeLdap(IO(ldapSearchStream(groupsOu, SearchScope.SUB, filters: _*)(unmarshalGroupThrow)))
+  }
+
+  def isUserMemberOfGroup(groupName: WorkbenchGroupName, userId: WorkbenchUserId): IO[Boolean] = {
+    val members = ldapLoadGroup(groupName).map(_.get.members)
+    val isDirectMember = members.map( _.contains(userId))
+    val isGroupMember = members.flatMap( _.collect{case x:WorkbenchGroupName => x}.toList.existsM(isUserMemberOfGroup(_, userId)))
+
+    List(isDirectMember, isGroupMember).sequence.map( _.contains(true) )
+  }
```

Comment:
this also needs a short-circuit - we don't want to query the subgroups if the user is a direct member. Something like ... 
```
  def isUserMemberOfGroup(groupName: WorkbenchGroupName, userId: WorkbenchUserId): IO[Boolean] =
    for {
      group <- ldapLoadGroup(groupName)
      members = group.map(_.members).getOrElse(Set.empty[WorkbenchSubject])
      isDirectMember = members.contains(userId)
      isGroupMember <- if (isDirectMember) IO.pure(false) else members.collect{case x:WorkbenchGroupName => x}.toList.existsM(isUserMemberOfGroup(_, userId))
    } yield {
      isDirectMember || isGroupMember
    }
```
(though I don't like the hardcoded `IO.pure(false)` for isGroupMember